src/Integration.Vsix.UnitTests/Analysis/IssueHandlerTests.cs

```diff
@@ -0,0 +1,271 @@
+            };
+
+            var issuesFilter = CreateIssuesFilter(out var issuesPassedToFilter, issuesToReturnFromFilter);
+            var publisher = new SnapshotPublisher();
+
+            var expectedGuid = Guid.NewGuid();
+            const string expectedProjectName = "my project name";
+            const string expectedFilePath = "c:\\aaa\\file.txt";
+            
+            var testSubject = CreateTestSubject(issuesFilter.Object, publisher.PublishSnapshot,
+                expectedProjectName, expectedGuid, expectedFilePath);
+
+            // Act
+            testSubject.HandleNewIssues(inputIssues);
+
+            // Assert
+            publisher.InvocationCount.Should().Be(1);
+
+            // Check the expected issues were passed to the filter
+            issuesPassedToFilter.Count.Should().Be(2);
+            issuesPassedToFilter.Should().BeEquivalentTo(inputIssues, c => c.WithStrictOrdering());
+
+            // Check the publish issues
+            publisher.PublishedSnapshot.Issues.Count().Should().Be(1);
+            publisher.PublishedSnapshot.Issues.First().Issue.RuleKey.Should().Be("xxx");
+
+            publisher.PublishedSnapshot.TryGetValue(0, StandardTableKeyNames.ProjectName, out var actualProjectName).Should().BeTrue();
+            actualProjectName.Should().Be(expectedProjectName);
+
+            publisher.PublishedSnapshot.TryGetValue(0, StandardTableKeyNames.ProjectGuid, out var actualProjectGuid).Should().BeTrue();
+            actualProjectGuid.Should().Be(expectedGuid);
+
+            publisher.PublishedSnapshot.TryGetValue(0, StandardTableKeyNames.DocumentName, out var actualFilePath).Should().BeTrue();
+            actualFilePath.Should().Be(expectedFilePath);
+
+            publisher.PublishedSnapshot.AnalyzedFilePath.Should().Be(expectedFilePath);
+        }
+
+        [TestMethod]
+        public void WhenNewIssuesAreFound_AndFilterRemovesAllIssues_ListenersAreUpdated()
+        {
+            // Arrange
+            var filteredIssue = CreateIssue("single issue", startLine: 1, endLine: 1);
+
+            var issuesToReturnFromFilter = Enumerable.Empty<IAnalysisIssueVisualization>();
+            var issuesFilter = CreateIssuesFilter(out var capturedFilterInput, issuesToReturnFromFilter);
+
+            var publisher = new SnapshotPublisher();
+            var testSubject = CreateTestSubject(issuesFilter.Object, publisher.PublishSnapshot);
+
+            // Act
+            testSubject.HandleNewIssues(new[] { filteredIssue });
+
+            // Assert
+            // Check the expected issues were passed to the filter
+            capturedFilterInput.Count.Should().Be(1);
+            capturedFilterInput[0].Should().Be(filteredIssue);
+
+            publisher.InvocationCount.Should().Be(1);
+            publisher.PublishedSnapshot.Issues.Count().Should().Be(0);
+        }
+
+        [TestMethod]
+        public void WhenNewFileLevelIssuesAreFound_IssueNotFiltered_ListenersAreUpdated()
```

Comment:
Again, it's an old test but.. Issue filtering is irrelevant here, no? It should be "HandleNewIssues_HasFileLevelIssue_IssueIsReturned"

It has nothing to do with filtering since a file level CAN be filtered out; we're simply returning all issues that have spans OR are file level. 
WDYT?